### PR TITLE
Reviews POST and GET /api/v1/reviews/

### PR DESCRIPTION
**Pull request by alect47**
- Creates reviews model and migration to allow user to POST and GET tacos from the database
- Created review serializer and model
- added average_rating property to Taco model
- User can retrieve a list of all the reviews currently in the database.
Endpoint: GET /api/v1/reviews
When I make the GET request, I expect to see 200 status code and the following response data in the body:
```
Body:
[
    {
        "id": 2,
        "rating": 4.0,
        "review": "test review",
        "taco": 27
    },
    {
        "id": 3,
        "rating": 4.0,
        "review": "test review",
        "taco": 27
    },
    {
        "id": 4,
        "rating": 4.0,
        "review": null,
        "taco": 27
    }
]
```
This will also change the GET /api/v1/tacos endpoint to include review information

```
body:
[
    {
        "id": 27,
        "type": "carne asada",
        "restaurant_id": 2361,
        "average_rating": 5.833333333333333,
        "reviews": [
            {
                "id": 2,
                "rating": 4.0,
                "review": "test review",
                "taco": 27
            },
            {
                "id": 3,
                "rating": 4.0,
                "review": "test review",
                "taco": 27
            }
          ]
       }
]
```
- User can post a new review to the database
- Endpoint: POST /api/v1/reviews
- Post request body should be formatted as follows:
```
{"rating": 10, "review": "is ok", "taco": 27}
```
- Response sent with 201 status code:
```
body:
{
    "id": 8,
    "rating": 10.0,
    "review": "is ok",
    "taco": 27
}
```

### CODE REVIEW

File: myapi/urls.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 router = routers.DefaultRouter()
 router.register(r'tacos', views.TacoViewSet)
 router.register(r'restaurants', views.RestaurantViewSet)
+router.register(r'reviews', views.ReviewViewSet)
```
Comment (scottzero) [inline myapi/urls.py:9]: Nice work adding these URLs in. Makes it so much easier to understand when they are uniform!

File: myapi/views.py [hunk 2]
```diff
@@ -13,9 +13,9 @@
 from rest_framework.response import Response
 
 from .serializers import HeroSerializer
-from .serializers import RestaurantSerializer, TacoSerializer
+from .serializers import RestaurantSerializer, TacoSerializer, ReviewSerializer
 
-from .models import Hero, Taco
+from .models import Hero, Taco, Review
 from .models import Restaurant
 # Create your views here.
```
Comment (scottzero) [inline myapi/views.py:21]: We can't forget to remove our unused imports later!

File: myapi/serializers.py [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
 
+class TacoSerializer(serializers.ModelSerializer):
+    reviews = ReviewSerializer(many=True, read_only=True)
+    average_rating = serializers.ReadOnlyField()
+
+    class Meta:
+        model = Taco
+        # the field reviews must be named exactly the same as the related name
+        fields = ('id', 'type', 'restaurant_id', 'average_rating', 'reviews')
+
```
Comment (scottzero) [inline myapi/serializers.py:25]: Front end was cool with removing vegan bool in the taco fields? (im cool with it)

File: tacoboutit/settings.py [hunk 4]
```diff
@@ -80,19 +80,19 @@
...
+# DATABASES = {
+#     'default': {
+#         'ENGINE': 'django.db.backends.postgresql',
+#         'NAME': 'tacoboutit',
+#         'USER': '',
+#         'PASSWORD': '',
+#         'HOST': '',
+#         'PORT': ''
+#     }
+# }
```
Comment (scottzero) [inline tacoboutit/settings.py:96]: Was this the solution to us having to "switch modes" every time we wanted to push to production? LOL

**Comment by scottzero** [comment]
Looks good, I left some comments above for specific changes. Thanks for the PR and nice work! :+1:

File: myapi/views.py (hunk 2 above)
Comment (alect47) [reply myapi/views.py:21]: Nice catch, I actually am working on removing them now

File: myapi/serializers.py (hunk 3 above)
Comment (alect47) [reply myapi/serializers.py:25]: Yeah, I asked about it and they decided to just make vegan a taco type rather than its own field

**Pull request merged**